spring-boot-modules/spring-boot-crud/src/main/java/com/baeldung/entity_dto_differences/dto/UserCreationDto.java

```diff
@@ -0,0 +1,54 @@
+package com.baeldung.entity_dto_differences.dto;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import javax.validation.constraints.NotNull;
+import java.util.List;
+
+public class UserCreationDto {
+
+    @NotNull
+    @JsonProperty("FIRST_NAME")
+    private final String firstName;
+
+    @NotNull
+    @JsonProperty("LAST_NAME")
+    private final String lastName;
+
+    @NotNull
+    @JsonProperty("ADDRESS")
+    private final String address;
+
+    @NotNull
+    @JsonProperty("BOOKS")
+    private final List<BookDto> books;
+
+    // Default constructor for Jackson deserialization
+    public UserCreationDto() {
+        this.firstName = null;
+        this.lastName = null;
+        this.address = null;
+        this.books = null;
+    }
+
+    public UserCreationDto(String firstName, String lastName, String address, List<BookDto> books) {
+        this.firstName = firstName;
+        this.lastName = lastName;
+        this.address = address;
+        this.books = books;
+    }
+
+    public String getFirstName() {
+        return firstName;
+    }
+
+    public String getLastName() {
+        return lastName;
+    }
+
+    public String getAddress() { return address; }
```

Comment:
Fix the formatting. Again, if you use the standard formatter you can just auto format not have to worry about the code guidelines.